spa/src/hooks/usePortalContribution.test.tsx

```diff
@@ -299,4 +300,50 @@ describe('usePortalContribution', () => {
       errorSpy.mockRestore();
     });
   });
+
+  describe('The sendEmailReceipt function', () => {
+    it('is returned even when a contribution is loading', async () => {
+      const { result } = hook(123, 1);
+
+      expect(typeof result.current.sendEmailReceipt).toBe('function');
+    });
```

Comment:
```suggestion
    it('is returned even when a contribution is loading', async () => {
      const { result, waitForNextUpdate } = hook(123, 1);

      expect(typeof result.current.sendEmailReceipt).toBe('function');
      await waitForNextUpdate();
    });
```

This lets the hook initial load finish properly; otherwise you get an intermittent "tried to update an unmounted component" error in test runs.